### PR TITLE
feat: add lint-staged

### PR DESCRIPTION
**Pull request by eugustavo**
Adding lint-staged to check the code linting before any commit.

### CODE REVIEW

**Comment by diego3g** [comment]
Does the `lint-staged` get triggered when the `git commit` comes from the root of the project instead from the `web` and `server` folders? If not, maybe it would be a good idea to keep the lint-staged in the root package.json instead of each package.

**Comment by eugustavo** [comment]
> Does the get triggered when the comes from the root of the project instead from the and folders? If not, maybe it would be a good idea to keep the lint-staged in the root package.json instead of each package.`lint-staged``git commit``web``server`

What came to my mind when putting each folder was that the ESLint rules are different. As each folder has its internal ESLint rules, I chose to have each one responsible for linting the code based on the rules that each one contains.

Putting all the rules together would not be a problem as some are just the Web and others are just the Server?

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+**/*.js
+/*.js
+node_modules
+build
+/src/react-app-env.d.ts
```
Comment (diego3g) [inline .eslintignore:5]: I think this path is wrong, we should add web folder to path.

File: prettier.config.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  singleQuote: true,
+  trailingComma: 'all',
+	arrowParens: 'avoid',
```
Comment (diego3g) [inline prettier.config.js:4]: I think there is extra space in this line.

Comment (eugustavo) [reply prettier.config.js:4]: How strange, there is no extra space in my code. But, I removed all the spaces and added them again.

**Pull request merged**